logic/resolvable/ResolvableFormula.java

```diff
@@ -0,0 +1,108 @@
+    public static class ResolvableDisjunction {
+        private final Disjunction disjunctionPattern;
+        private final Set<ResolvableConjunction> resolvableConjunctions;
+
+        private ResolvableDisjunction(Disjunction disjunctionPattern, Set<ResolvableConjunction> conjunctions) {
+            this.disjunctionPattern = disjunctionPattern;
+            this.resolvableConjunctions = conjunctions;
+        }
+
+        public static ResolvableDisjunction of(Disjunction disjunctionPattern) {
+            Set<ResolvableConjunction> resolvableConjunctions = iterate(disjunctionPattern.conjunctions())
+                    .map(c -> ResolvableConjunction.of(c)).toSet();
+            return new ResolvableDisjunction(disjunctionPattern, resolvableConjunctions);
+        }
+
+        public Disjunction pattern() {
+            return disjunctionPattern;
+        }
+
+        public Set<ResolvableConjunction> conjunctions() {
+            return resolvableConjunctions;
+        }
+    }
+
+    public static class ResolvableConjunction {
+        private final Conjunction conjunctionPattern;
+        private final Set<ResolvableNegation> negations;
+        private final Set<Concludable> concludables;
+
+        private ResolvableConjunction(Conjunction conjunction, Set<Concludable> concludables, Set<ResolvableNegation> negations) {
+            this.conjunctionPattern = conjunction;
+            this.negations = negations;
+            this.concludables = concludables;
+        }
+
+        public static ResolvableConjunction of(Conjunction conjunction) {
+            Set<Concludable> concludables = Concludable.create(conjunction);
+            Set<ResolvableNegation> negations = iterate(conjunction.negations())
+                    .map(neg -> ResolvableNegation.of(neg))
+                    .toSet();
+
+            return new ResolvableConjunction(conjunction, concludables, negations);
+        }
+
+        public Conjunction pattern() {
+            return conjunctionPattern;
+        }
+
+        public Set<ResolvableNegation> negations() {
+            return negations;
+        }
+
+        public Set<Concludable> concludables() {
+            return concludables;
+        }
+    }
+
+    public static class ResolvableNegation {
+        private final Negation negationPattern;
+        private final ResolvableDisjunction disjunction;
+
+        public ResolvableNegation(Negation negationPattern, ResolvableDisjunction resolvableDisjunction) {
+            this.negationPattern = negationPattern;
+            this.disjunction = resolvableDisjunction;
+        }
+
+        public static ResolvableNegation of(Negation negation) {
+            return new ResolvableNegation(negation, ResolvableDisjunction.of(negation.disjunction()));
+        }
+
+        public ResolvableDisjunction disjunction() {
+            return disjunction;
+        }
+
+        public Negation pattern() {
+            return negationPattern;
+        }
+    }
```

Comment:
Define a ResolvableNegation to

1. be derived from a pattern.Negation
2. Consist of a ResolvableDisjunction